superset-frontend/src/features/tags/tags.ts

```diff
@@ -163,9 +164,6 @@ export function addTag(
   if (objectType === undefined || objectId === undefined) {
     throw new Error('Need to specify objectType and objectId');
   }
-  if (tag.indexOf(':') !== -1 && !includeTypes) {
-    return;
-  }
```

Comment:
This was preventing tags with `:` to be successfully added to assets. Given that system tags are not included in the Tags list (in the Properties modal) we should be good to remove this check.